### DISCUSSION TITLE
How does the zfs-autobackup tool find the settings that tell it what zfs fs to check for snapshotting?

### DISCUSSION THREAD

**Discussion by hp3-x** [psy0rz/zfs_autobackup#281] [Q&A]
Trying to understand the mechanics of how zfs-autobackup tool operates (referred to as "zab" from here onward)   Does the zab tool do something similar to grepping the output of 'zfs get all pool/zfsfs'.
Or something less labor intensive? To learn what it is expected to do?
I'd like to be able to just read the code and understand what happens, but there is no chance at all that I can do that yet.  Hopefully someone is willing to explain in plain English how "zab" gets its marching orders.  Just on the simplest level as in using it to snapshot only.

**Comment by psy0rz** [comment]
Please read the full manual at the wiki. It should explain all that is happening.

**Comment by psy0rz** [comment]
Here: https://github.com/psy0rz/zfs_autobackup/wiki/Manual#backup-procedure

**Reply by hp3-x** [reply]
Thanks for the pointer directly do the related manual parts.
Sequence of events will be good info to have.

However I asked about something a little more basic.  How zfs-autobackup actually finds the bits in the possible moderators that tell it what to do.  I asked if the python code does something along the line of  zfs get all ... | grep ... That is, something that does similar in python.... how does the code scan for the needed settings?

**Reply by psy0rz** [reply]
Ah you mean this: https://github.com/psy0rz/zfs_autobackup/blob/81fa5c5bab29d9d7a3b9145602f0395de8093cd7/zfs_autobackup/ZfsNode.py#L242
